### PR TITLE
Backup: Escape DB name in checkNoDB().

### PR DESCRIPTION
**Pull request by enisoc**
Ran into this with a keyspace that has `-` in the name.

#3620

### CODE REVIEW

File: go/vt/mysqlctl/backup.go [hunk 1]
```diff
@@ -539,9 +540,10 @@ func checkNoDB(ctx context.Context, mysqld MysqlDaemon, dbName string) (bool, er
 		return false, fmt.Errorf("checkNoDB failed: %v", err)
 	}
 
+	backtickDBName := sqlescape.EscapeID(dbName)
 	for _, row := range qr.Rows {
 		if row[0].ToString() == dbName {
```
Comment (derekperkins) [inline go/vt/mysqlctl/backup.go]: Will this match now that it's backticked?

Comment (derekperkins) [reply go/vt/mysqlctl/backup.go]: Never mind, it's still the same comparison, my bad.

**Comment by derekperkins** [comment]
LGTM - #3620

**Comment by alainjobart** [comment]
LGTM

**Pull request merged**